ArduCopter/mode_auto.cpp

```diff
@@ -1490,7 +1490,14 @@ void Copter::ModeAuto::do_RTL(void)
 bool Copter::ModeAuto::verify_takeoff()
 {
     // have we reached our target altitude?
-    return copter.wp_nav->reached_wp_destination();
+    bool reached_wp_dest = copter.wp_nav->reached_wp_destination();
+
+    // retract the landing gear
+    if (SRV_Channels::function_assigned(SRV_Channel::k_landing_gear_control) && reached_wp_dest) {
```

Comment:
We shouldn't make the assumption that the landing gear is triggered by an rc output channel.  We should just assuming the landing gear function will ignore the command if no output channel is defined.

Reply:
> We shouldn't make the assumption that the landing gear is triggered by an rc output channel. We should just assuming the landing gear function will ignore the command if no output channel is defined.

How should I update this part?

Reply:
@liang-tang, just remove the "function_assigned" check completely.

Reply:
like this "if (reached_wp_dest) { "?

Reply:
@liang-tang, yes, that sounds right

Reply:
done!